Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SyncKronosToShiftsController.cs

```diff
@@ -104,51 +104,122 @@ public ActionResult SyncKronosToShifts(string isRequestFromLogicApp)
-
+                isMapPayCodeTimeOffReasonsSuccessful = true;
                 this.telemetryClient.TrackTrace($"{Resource.MapPayCodeTimeOffReasonsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                isMapPayCodeTimeOffReasonsSuccessful = false;
+                this.telemetryClient.TrackException(ex);
+            }
 
-                // Sync timeoff from kronos to shifts.
-                await this.timeOffController.ProcessTimeOffsAsync(isRequestFromLogicApp).ConfigureAwait(false);
-
-                this.telemetryClient.TrackTrace($"{Resource.ProcessTimeOffsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
-
-                // Sync timeoff from kronos to shifts.
-                await this.timeOffRequestsController.ProcessTimeOffRequetsAsync(isRequestFromLogicApp).ConfigureAwait(false);
-
-                this.telemetryClient.TrackTrace($"{Resource.SyncTimeOffRequestsFromShiftsToKronos} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+            // Sync TimeOff and TimeOffRequest only if Paycodes in kronos synced successfully.
+             if (isMapPayCodeTimeOffReasonsSuccessful)
+            {
+                try
+                {
+                    // Sync timeoff from kronos to shifts.
+                    await this.timeOffController.ProcessTimeOffsAsync(isRequestFromLogicApp).ConfigureAwait(false);
+
+                    this.telemetryClient.TrackTrace($"{Resource.ProcessTimeOffsAsync} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+                }
+#pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                {
+                    this.telemetryClient.TrackException(ex);
+                }
+
+                try
+                {
+                    // Sync timeoff from kronos to shifts.
+                    await this.timeOffRequestsController.ProcessTimeOffRequestsAsync(isRequestFromLogicApp).ConfigureAwait(false);
+
+                    this.telemetryClient.TrackTrace($"{Resource.SyncTimeOffRequestsFromShiftsToKronos} completed from {Resource.ProcessKronosToShiftsShiftsAsync} ");
+                }
+#pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                {
+                    this.telemetryClient.TrackException(ex);
+                }
+            }
+            else
+            {
+                this.telemetryClient.TrackTrace($"{Resource.MapPayCodeTimeOffReasonsAsync} status:  " + isMapPayCodeTimeOffReasonsSuccessful);
```

Comment:
Change this to actually describe what's happening, that you skipped processing time offs and time off requests because XXX.

Reply:
@aosolis - ack'd and changing the telemetry accordingly.